### PR TITLE
Bugfix FXIOS-5550 [v114] Crash  UIKitCore: -[UICollectionView _endItemAnimationsWithInvalidationContext...]

### PR DESCRIPTION
**Pull request by yoanarios**
[Jira ticket](https://mozilla-hub.atlassian.net/browse/FXIOS-5550)
[Github issue](https://github.com/mozilla-mobile/firefox-ios/issues/12871)

### Description
- Remove unnecessary calls to performBatchUpdate
- Move dataSource and collectionView changes into `updateWith` completion which is called from `performBatchUpdates` update closure as recommended from Apple's doc

### Pull requests checks where applicable
- [x] Fill in the three TODOs above (tickets number and description of your work)
- [x] The PR name follows our [naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [ ] Accessibility implemented and tested (minimum Dynamic Text and VoiceOver)
- [ ] Unit tests written and passing
- [x] Documentation / comments for complex code and public methods

### CODE REVIEW

File: Client/Frontend/Browser/TabDisplayManager.swift [hunk 1]
```diff
@@ -1078,9 +1087,7 @@ extension TabDisplayManager: TabManagerDelegate {
         /// collectionView is not visible the dataSource section/items differs from the actions to be perform
         /// which causes the crash
         collectionView.numberOfItems(inSection: 0)
-        collectionView.performBatchUpdates({ [weak self] in
-            // Baseline animation speed is 1.0, which is too slow, this (odd) code sets it to 3x
-            self?.collectionView.forFirstBaselineLayout.layer.speed = 3.0
```
Comment (nishant2718) [inline Client/Frontend/Browser/TabDisplayManager.swift:1083]: Removing this is intentional?

File: Client/Frontend/Browser/TabDisplayManager.swift [hunk 2]
```diff
@@ -889,16 +885,19 @@ extension TabDisplayManager: UICollectionViewDropDelegate {
             saveRegularOrderedTabs(from: filteredTabs)
         }
 
-        dataStore.removeAll()
+        /// According to Apple's documentation the best place to make the changes to the collectionView and dataStore is
+        /// during the completion call of performBatchUpdates
+        updateWith(animationType: .moveTab) { [weak self] in
```
Comment (nishant2718) [inline Client/Frontend/Browser/TabDisplayManager.swift:890]: 👌

File: Client/Frontend/Browser/TabDisplayManager.swift (hunk 1 above)
Comment (yoanarios) [reply Client/Frontend/Browser/TabDisplayManager.swift:1083]: Yes `forFirstBaselineLayout` is a property F`or views with multiple rows of text` as per Apple's documentation so I couldn't find anything justifying this line of code

**Comment by yoanarios** [comment]
Green build
Screenshot 2023-04-25 at 9 28 28 AM

**Review by nishant2718 — APPROVED** [review]

**Pull request merged**